### PR TITLE
Adding DDB BillingMode Support

### PR DESCRIPTION
**Pull request by Cory-Bradshaw**
*Description of changes:*
Adding BillingMode Support to KCL 1.x. This enables the customer to specify if they want provisioned capacity for DDB, or pay per request.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncTaskIntegrationTest.java [hunk 1]
```diff
@@ -99,25 +103,79 @@ public void setUp() throws Exception {
...
+        }
+    }
+
+    private void checkBillingMode(BillingMode billingMode, String tableName) {
+        AmazonDynamoDBClient client = new AmazonDynamoDBClient(DefaultAWSCredentialsProviderChain.getInstance());
+        DescribeTableResult tableDetails = client.describeTable(tableName);
+        if(BillingMode.PAY_PER_REQUEST.equals(billingMode)) {
+            Assert.assertEquals(tableDetails.getTable().getBillingModeSummary().getBillingMode(), billingMode.name());
+        }else{
+            Assert.assertTrue(tableDetails.getTable().getProvisionedThroughput().getWriteCapacityUnits() == 10);
```
[the remark this answers is not in the document]
Comment (Cory-Bradshaw) [reply src/test/java/com/amazonaws/services/kinesis/clientlibrary/lib/worker/ShardSyncTaskIntegrationTest.java]: Why not assertEquals here?
==> equals doesn't work on value types, didn't seem more ugly than casting to Integer

Billing modes are controlled by the billing modes we use to create the table with. If DDB adds a new billing mode, it wouldn't effect this check unless we change the unit test to use a different billing mode.

File: src/test/java/com/amazonaws/services/kinesis/leases/impl/LeaseManagerIntegrationTest.java [hunk 2]
```diff
@@ -233,7 +234,24 @@ public void testDeleteNonexistentLease() throws LeasingException {
...
+
+        };
+
+        Assert.assertTrue(manager.waitUntilLeaseTableExists(1, 1));
+    }
+
+    @Test
+    public void testWaitUntilLeaseTableExistsPayPerRequest() throws LeasingException {
+        KinesisClientLeaseManager manager = new KinesisClientLeaseManager("nagl_ShardProgress_PayPerRequest", ddbClient, true,
+                KinesisClientLibConfiguration.DEFAULT_DDB_BILLING_MODE) {
```
[the remark this answers is not in the document]
Comment (Cory-Bradshaw) [reply src/test/java/com/amazonaws/services/kinesis/leases/impl/LeaseManagerIntegrationTest.java]: Good Catch. Updating..

File: src/main/java/com/amazonaws/services/kinesis/leases/impl/KinesisClientLeaseManager.java [hunk 3]
```diff
@@ -38,8 +39,8 @@
      * @param table Leases table
      * @param dynamoDBClient DynamoDB client to use
      */
-    public KinesisClientLeaseManager(String table, AmazonDynamoDB dynamoDBClient) {
-        this(table, dynamoDBClient, false);
+    public KinesisClientLeaseManager(String table, AmazonDynamoDB dynamoDBClient, BillingMode billingMode) {
```
[the remark this answers is not in the document]
Comment (Cory-Bradshaw) [reply src/main/java/com/amazonaws/services/kinesis/leases/impl/KinesisClientLeaseManager.java]: Discussed offline, both methods have their own cost. Going to go with adding fields to existing constructors over continuing to increase the number of constructors.

[the remark this answers is not in the document]
Comment (jirizaloudek) [reply src/main/java/com/amazonaws/services/kinesis/leases/impl/KinesisClientLeaseManager.java]: Sorry, but I do not agree. You have just break the backward compatibility and every client using your library must upgrade it's source code.

In my case, the new 1.13.1 version was downloaded by Maven (I did not fixed the version in dependency management, my fault). But the dynamodb-streams-kinesis-adapter-1.5.0.jar is using the old 2 parameters constructor and starts to fail. So I need manually freeze amazon-kinesis-client-1.13.0.jar in Maven and wait for a new version of dynamodb-streams-kinesis-adapter that will be compatible with amazon-kinesis-client-1.13.1

[the remark this answers is not in the document]
Comment (jirizaloudek) [reply src/main/java/com/amazonaws/services/kinesis/leases/impl/KinesisClientLeaseManager.java]: I find there is already open an issue https://github.com/awslabs/dynamodb-streams-kinesis-adapter/issues/27

**Pull request merged**